Sources/StreamChat/Database/DTOs/UserDTO.swift

```diff
@@ -30,10 +33,24 @@ class UserDTO: NSManagedObject {
         request.predicate = NSPredicate(format: "id == %@", userId)
         return request
     }
-    
+
     override func willSave() {
         super.willSave()
-        
+
+        // When user changed, we need to propagate this change to members and current user
+        if hasPersistentChangedValues {
+            if !(currentUser?.hasChanges ?? false) {
+                // this will not change object, but mark it as dirty, triggering updates
+                currentUser?.unreadChannelsCount = currentUser?.unreadChannelsCount ?? 0
+            }
+            for member in members ?? [] {
+                guard !member.hasChanges else { continue }
+                // this will not change object, but mark it as dirty, triggering updates
+                let swiftBelieveMeIKnowWhatImDoing = member.channelRoleRaw
+                member.channelRoleRaw = swiftBelieveMeIKnowWhatImDoing
```

Comment:
I understand this is a workaround for the compiler warning. I wouldn't mind if you named it the way such that it's more obvious.